bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/dto/clip2/helper/Setters.java

```diff
@@ -0,0 +1,304 @@
+import org.openhab.core.library.types.HSBType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.types.Command;
+import org.openhab.core.util.ColorUtil;
+import org.openhab.core.util.ColorUtil.Gamut;
+
+/**
+ * Advanced setter methods for fields in the Resource class for special cases where setting the new value in the target
+ * resource depends on logic using the values of other fields in a another source Resource.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ */
+@NonNullByDefault
+public class Setters {
+
+    /**
+     * Setter for Alert field:
+     * Use the given command value to set the target resource DTO value based on the attributes of the source resource
+     * (if any).
+     *
+     * @param target the target resource.
+     * @param command the new state command should be a StringType.
+     * @param source another resource containing the allowed alert action values.
+     *
+     * @return the target resource.
+     */
+    public static Resource setAlert(Resource target, Command command, @Nullable Resource source) {
+        if ((command instanceof StringType) && Objects.nonNull(source)) {
+            Alerts otherAlert = source.getAlerts();
+            if (Objects.nonNull(otherAlert)) {
+                ActionType actionType = ActionType.of(((StringType) command).toString());
+                if (otherAlert.getActionValues().contains(actionType)) {
+                    target.setAlerts(new Alerts().setAction(actionType));
+                }
+            }
+        }
+        return target;
+    }
+
+    /**
+     * Setter for Color Temperature field:
+     * Use the given command value to set the target resource DTO value based on the attributes of the source resource
+     * (if any).
+     *
+     * @param target the target resource.
+     * @param command the new state command should be a QuantityType<Temperature> (but it can also handle DecimalType).
+     * @param source another resource containing the MirekSchema.
+     *
+     * @return the target resource.
+     */
+    public static Resource setColorTemperatureAbsolute(Resource target, Command command, @Nullable Resource source) {
+        QuantityType<?> colorTemperatureCommand = null;
+        if (command instanceof QuantityType<?>) {
+            colorTemperatureCommand = (QuantityType<?>) command;
+        } else if (command instanceof DecimalType) {
+            colorTemperatureCommand = QuantityType.valueOf(((DecimalType) command).doubleValue(), Units.KELVIN);
+        }
+        QuantityType<?> kelvin = Objects.nonNull(colorTemperatureCommand)
+                ? colorTemperatureCommand.toInvertibleUnit(Units.KELVIN)
+                : null;
+        if (Objects.nonNull(kelvin)) {
+            QuantityType<?> mirek = kelvin.toInvertibleUnit(Units.MIRED);
```

Comment:
Why convert to Kelvin and then convert again to Mired? Please consider this for the suggestion above, could it be converted to Mired directly?

Reply:
See above.